README.md

```diff
@@ -144,80 +144,115 @@ The old repo is no longer the recommended approach for running and extending the
+| `MECH_TO_CONFIG`           | `dict` | `{"0x895c50590a516b451668a620a9ef9b8286b9e72d":{"use_dynamic_pricing":false,"is_marketplace_mech":false}}`                                                                                                                                                          | Tracks mech's config.                                                  |
+| `MECH_TO_SUBSCRIPTION`     | `dict` | `{"0x895c50590a516b451668a620a9ef9b8286b9e72d":{"tokenAddress":"0x0000000000000000000000000000000000000000","tokenId":"1"}}`                                                                                                                                        | Tracks mech's subscription details.                                    |
 
+:warning: The addresses in the variables `MECH_TO_CONFIG` and `MECH_TO_SUBSCRIPTION` should be identical and correspond 
+to the address of the Mech contract. Furthermore, `NUM_AGENTS` has to be between 1 and 4.
 
-> **Warning**<br />
-> **The demo service is configured to match a specific on-chain agent (ID 3 on [Mech Hub](https://aimechs.autonolas.network/registry)). Since you will not have access to its private key, your local instance will not be able to transact.
-> However, it will be able to receive Requests for AI tasks [sent from Mech Hub](https://aimechs.autonolas.network/mech). These Requests will be executed by your local instance, but you will notice that a failure will occur when it tries to submit the transaction on-chain (Deliver type).**
+If you want to run a legacy Mech, the `MECH_MARKETPLACE_ADDRESS` is optional. Otherwise this variable needs to be defined. 
+Furthermore, in the variable `MECH_TO_CONFIG`, the value corresponding to the key `is_marketplace_mech` should be set to true.
 
-Now, you have two options to run the worker: as a standalone agent or as a service.
+In order to run your custom tool, you need to add its name and hash to the variable `TOOLS_TO_PACKAGE_HASH`.
 
-### Option 1: Run the Mech as a standalone agent
+When running the Mech as a service, follow these additional steps: 
 
-1. Ensure you have a file with a private key (`ethereum_private_key.txt`). You can generate a new private key file using the Open Autonomy CLI:
+- Ensure that the variable `ALL_PARTICIPANTS` in the file `.1env` contains the same agent instance address as in `keys.json` when running the Mech as a service.
 
-   ```bash
-   autonomy generate-key ethereum 
-   ```
+- In the file `run_service.sh`, change the line `[autonomy deploy build -ltm](https://github.com/valory-xyz/mech-predict/blob/main/run_service.sh#L28)` to `autonomy deploy build -ltm -n k`,
+where k is replaced by the number of agents in your service (value of the variable `NUM_AGENTS`). For instance for 1 agent: `autonomy deploy build -ltm -n 1`.
```

Comment:
Why not use the `NUM_AGENTS` value in the script and set it on the user's behalf?